doc/on-behalf-of.md

```diff
@@ -0,0 +1,53 @@
+# On-Behalf-Of Authorization
+
+⚠️ Only compatible with Grafana 8.3.4 or later above.
+
+The feature can be enabled with the checkbox from the plugin configuration screen.
+When provisioning, you also need to set `oauthPassThru` to `true`, such as the following example has done.
+
+```yaml
+apiVersion: 1
+datasources:
+  - name: Azure Data Explorer
+    type: grafana-azure-data-explorer-datasource
+    access: proxy
+    basicAuth: false
+    editable: true
+    jsonData:
+      onBehalfOf: true
+      oauthPassThru: true
+      clientId: <your client UUID>
+      clusterUrl: <your cluster URL>
+      tenantId: <your tenant UUID>
+      tlsAuth: false
+      tlsAuthWithCACert: false
+      defaultDatabase: <your default database>
+    secureJsonData:
+      clientSecret: <your client secret>
+      tlsCACert: ''
+      tlsClientCert: ''
+      tlsClientKey: ''
+    version: 1
+```
+
+**note***: Grafana alert-rules do not work with on-behalf-of authorization.
```

Comment:
After closer consideration, this is actually not fully accurate. Please change this to:
_It is not recommended to setup [alerts](https://grafana.com/docs/grafana/latest/alerting/) when the data source is configured to use on-behalf-of authorization. This is because the alert rule will no longer work as soon as the user that created the rule is logged out from Grafana._